### PR TITLE
Bugfixes

### PR DESCRIPTION
**Pull request by alfiopuglisi**
Fixes for various bugs found mixing GPU and CPU

### CODE REVIEW

File: pyssata/base_time_obj.py
Comment (FabioRossiArcetri) [inline pyssata/base_time_obj.py:1]: This should be >= than 0

**Comment by FabioRossiArcetri** [comment]
also note the conflicts

File: pyssata/base_time_obj.py
Comment (alfiopuglisi) [reply pyssata/base_time_obj.py:1]: Fixed, thanks

**Review by FabioRossiArcetri — APPROVED** [review]
ok now

**Pull request merged**